pkg/minikube/bootstrapper/kubeadm/kubeadm.go

```diff
@@ -254,11 +256,12 @@ func (k *Bootstrapper) init(cfg config.ClusterConfig) error {
 		}
 		return errors.Wrap(err, "wait")
 	}
+	kw.Close()
```

Comment:
I think this close should be after Wait

Reply:
When we get to this line, all the logs are in the pipeline but may not have finished being processed, so we close the pipeline. Closing the pipeline doesn't kill it, instead it writes `EOF` to the pipe and then prevents new writes. Once the pipeline is emptied it hits the `EOF` and calls `wg.Done()` (line 322) and then the wg.Wait() on the next line is unblocked.

tl;dr: If they were in the other order it would block forever because the pipeline processing would be waiting for a `EOF` which is only added on `kw.Close()`.